pkg/metrics/metrics.go

```diff
@@ -383,6 +385,54 @@ the maximum possible share value.`,
 	)
 )
 
+type LocalQueueMetricsConfig struct {
+	localQueueSelector labels.Selector
+}
+
+var lqMetricsConfigSingleton *LocalQueueMetricsConfig
```

Comment:
Would it be straightforward to capture the state in a new type, which is used by callers? Rather than using global state

Reply:
I did give this a shot at first but found it cumbersome

Reply:
@gabesaba thanks for pointing out. Indeed, I would suggest to revisit this @KPostOffice . We may have some examples of global state in the code base, but this is a code smell, and should not be taken lightly.

So, give it another shot and summarize in a comment what are the complications. I will hopefully have a bit of time tomorrow to wrap my head around the complications and hopefully guide it, until then, please revisit this.